### DISCUSSION TITLE
so-allow not allowing 1514 for Wazuh agent

### DISCUSSION THREAD

**Discussion by Cra5hedC0w** [Security-Onion-Solutions/securityonion#1638] [Unsupported Versions] [answered]
When running so-allow to allow a subnet for wazuh agent connectivity, the logs in the agent still indicate that 1514 is refused by the SO instance. Issuing _iptables -L |grep 1514_ returns nothing. Upon disabling firewalld, the agent can then connect. 

Please let me know if there is a workaround for this, or if this is a known issue.

**Comment by weslambert** [comment]
What is the output of the following?

`so-firewall includedhosts wazuh_endpoint`

**Reply by congnguyen1420** [reply]
I have the same problem, when I run this comand I have 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/98400336/9ef1874e-c4f1-4dd3-9602-870d6e193abd)

**Reply by dougburks** [reply]
@NguyenCong-14 This is an old discussion from 2020 and it has been marked as answered. Instead of replying to old discussions, please start a new discussion with detailed information about your system.

**Comment by netsecninja** [comment, marked as answer]
I'm on 2.3.2 and having a similar problem with the registration port, 1515. Same issue, both 1515 and 1514 are not listed in the firewall with _iptables -L | grep -E "151[4|5]_". My _so-firewall includedhosts wazuh_endpoint_ shows 10.0.0.0/8, which is accurate. I've confirmed with network pcaps the traffic is leaving my host, hitting my network firewall, and routed on to the SO box. But a tcpdump on SO shows no packets.

**Reply by weslambert** [reply]
Try running with -nL, so you are not trying to resolving the raw port to a common service.

**Reply by netsecninja** [reply]
Good suggestion. And FYI, I just reinstalled last night using the 2.3.2 ISO to deal with some other issues I was having as well as avoiding any complications with upgrades. While it shows a rule for TCP 1515, the connection issue still persists:

**ACCEPT     tcp  --  0.0.0.0/0            172.17.0.11          tcp dpt:1515**
ACCEPT     tcp  --  0.0.0.0/0            172.17.0.11          tcp dpt:1514
ACCEPT     udp  --  0.0.0.0/0            172.17.0.11          udp dpt:1514
**ACCEPT     tcp  --  10.0.0.0/8           0.0.0.0/0            tcp dpt:1515**
ACCEPT     udp  --  10.0.0.0/8           0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  10.0.0.0/8           0.0.0.0/0            tcp dpt:1514
ACCEPT     tcp  --  10.0.3.3             0.0.0.0/0            tcp dpt:1515
ACCEPT     udp  --  10.0.3.3             0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  10.0.3.3             0.0.0.0/0            tcp dpt:1514
ACCEPT     tcp  --  127.0.0.1            0.0.0.0/0            tcp dpt:1515
ACCEPT     udp  --  127.0.0.1            0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  127.0.0.1            0.0.0.0/0            tcp dpt:1514

**Reply by netsecninja** [reply]
And for reference, here's the tcpdump output from SO:

tcpdump -nni eth0 port 1515
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
21:10:19.371231 IP 10.0.2.20.59958 > 10.0.3.3.1515: Flags [S], seq 1245805100, win 64240, options [mss 1460,sackOK,TS val 2259604420 ecr 0,nop,wscale 7], length 0
21:10:19.371314 IP 10.0.3.3.1515 > 10.0.2.20.59958: Flags [R.], seq 0, ack 1245805101, win 0, length 0

**Reply by weslambert** [reply]
Let me see if I can duplicate on my side.

**Reply by weslambert** [reply]
What version agent/server are you running?

**Reply by netsecninja** [reply]
Agent - 3.13.2-1
Server - 3.13.1-1

When I run `/var/ossec/bin/agent-auth -m <manager_IP>` (replaced with SO IP) from my workstation, it sits there and does nothing. If I run `so-wazuh-agent-manage` on SO and list the agents, only SO is listed.

**Reply by weslambert** [reply]
Looks like the agent is newer than the server.  Have you tried the 3.13.1-1 agent?

**Reply by netsecninja** [reply]
That is a patch-level difference, and would not cause the wazuh manager to not respond to TCP connections. However, for completeness, on my Linux Mint endpoint I removed the old version, and installed the 3.13.1-1 with `sudo apt install wazuh-agent=3.13.1-1`. I then ran `sudo apt hold wazuh-agent` to stop the package from being upgraded in the future.
As expected, even after this when I try to register from the endpoint, I still get the same error:
```
$ sudo /var/ossec/bin/agent-auth -m 10.0.3.3
2020/11/03 09:31:38 agent-auth: INFO: Started (pid: 640315).
2020/11/03 09:31:38 agent-auth: INFO: Starting enrollment process to server: 10.0.3.3
2020/11/03 09:31:38 agent-auth: ERROR: Unable to connect to 10.0.3.3:1515
```
This isn't an issue with agent/server versions. The server is not even accepting TCP connections on port 1515. For reference, here's the registration documentation I am following: https://documentation.wazuh.com/3.13/user-manual/registering/index.html#simple-registration-service

**Reply by weslambert** [reply]
Have you confirmed authd is running on the box? 

`ps aux | grep ossec`
`ps aux | grep authd`

**Reply by netsecninja** [reply]
Good call. I do see ossec but not authd.
```
# ps aux | grep ossec
root      1270  0.0  0.0  32848  1220 ?        Sl   Nov02   0:05 /var/ossec/bin/ossec-execd
ossec     1602  0.0  0.0 254448  4284 ?        Sl   Nov02   1:08 /var/ossec/bin/ossec-agentd
root      1898  0.1  0.0 330064  5156 ?        SNl  Nov02   3:23 /var/ossec/bin/ossec-syscheckd
root      2728  0.0  0.0 401488  1628 ?        Sl   Nov02   0:42 /var/ossec/bin/ossec-logcollector
root      3393  0.4  0.0 272504 13136 ?        Sl   Nov02  11:25 /var/ossec/bin/wazuh-modulesd
root     13785  0.0  0.0 112812   936 pts/0    S+   18:57   0:00 grep --color=auto ossec
ossec    26546  0.0  0.0 715028  7640 ?        Sl   Nov02   1:07 /var/ossec/bin/wazuh-db
root     26636  0.0  0.0  43232  1456 ?        Sl   Nov02   0:02 /var/ossec/bin/ossec-execd
ossec    26667  0.0  0.1 1417308 24056 ?       Sl   Nov02   0:17 /var/ossec/bin/ossec-analysisd
root     26812  0.0  0.0 273736  6380 ?        SNl  Nov02   0:19 /var/ossec/bin/ossec-syscheckd
ossecr   26946  0.2  0.0 1175956 4316 ?        Sl   Nov02   3:36 /var/ossec/bin/ossec-remoted
root     27085  0.0  0.0 411864  1876 ?        Sl   Nov02   0:21 /var/ossec/bin/ossec-logcollector
ossec    27112  0.0  0.0  43288  2084 ?        Sl   Nov02   0:05 /var/ossec/bin/ossec-monitord
root     27264  0.0  0.0 338252  3800 ?        Sl   Nov02   0:37 /var/ossec/bin/wazuh-modulesd
ossec    27487  0.0  0.2 981156 36556 ?        Sl   Nov02   0:02 node /var/ossec/api/app.js

# ps aux | grep authd
root     13837  0.0  0.0 112808   936 pts/0    S+   18:57   0:00 grep --color=auto authd
```
Isn't authd something that should be running by default? I ran `whereis ossec-authd`, and it's not even installed.

**Reply by weslambert** [reply]
It would be inside of the Docker container.  

Try this:

`sudo docker exec so-wazuh /var/ossec/bin/ossec-authd`

If that doesn't work, you can go into the container and execute `ossec-authd` with:

```
sudo docker exec -it so-wazuh /bin/bash
cd /var/ossec/bin
./ossec-auth
exit
```

Then, from your remote machine, try `nc -vz <ip> 1515` to test the connection, or try running the agent join command again.

**Reply by netsecninja** [reply]
Progress! The first command worked to start the service. On my endpoint I ran `sudo /var/ossec/bin/agent-auth -m <manager_IP>`, which ran without error. I verified the endpoint was listed in SO with `so-wazuh-agent-manage`. 

So I assume if I reboot my SO box, that ossec-authd service will no longer be running. My question is, is it supposed to be?

**Reply by weslambert** [reply]
Yes, this was just a test.  I've put in a fix for this, which will be tracked here: https://github.com/Security-Onion-Solutions/securityonion/issues/1810.  The next release should have the fix.

**Reply by netsecninja** [reply]
Thanks for your help tracking this down Wes!

**Reply by kiaraho** [reply]
> It would be inside of the Docker container.
> 
> Try this:
> 
> `sudo docker exec so-wazuh /var/ossec/bin/ossec-authd`
> 
> If that doesn't work, you can go into the container and execute `ossec-authd` with:
> 
> ```
> sudo docker exec -it so-wazuh /bin/bash
> cd /var/ossec/bin
> ./ossec-auth
> exit
> ```
> 
> Then, from your remote machine, try `nc -vz <ip> 1515` to test the connection, or try running the agent join command again.

I also ran into the same `ossec-authd` not running issue as @netsecninja. The temporary fix above also worked for me, once I found this post after hours of troubleshooting. 🤦‍♂️

**Reply by manuelplascencia** [reply]
This was helpful thank you. Not sure if you answered netsecninja's question. Will a reboot require this to be ran ed again ?

**Comment by ozzbrian** [comment]
When running so-allow to allow a subnet for wazuh agent connectivity, the logs in the  indicate that 1514 is activialy refused by the SO instance. Issuing iptables -L |grep 1514 returns nothing. Upon disabling firewalld, the agent can then connect.

**Reply by weslambert** [reply]
Try running with `-nL`, so you are not trying to resolving the raw port to a common service

**Comment by Cra5hedC0w** [comment]
iptables -nL |grep 1514 shows (redacted IP/Mask): 

ACCEPT     tcp  --  0.0.0.0/0            172.17.0.9           tcp dpt:1514
ACCEPT     udp  --  0.0.0.0/0            172.17.0.9           udp dpt:1514
ACCEPT     udp  --  10.x.x.x/xx      0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  10.x.x.x.x/xx       0.0.0.0/0            tcp dpt:1514
ACCEPT     udp  --  127.0.0.1            0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  127.0.0.1            0.0.0.0/0            tcp dpt:1514


The logs for the agent still shows connection refused. I do see resets in the tcpdump. The rule exists in the DOCKER USER Chain. 
The port is labeled as "fujitsu-dtcns" for anyone else searching iptables.

**Comment by TOoSmOotH** [comment]
What version are you using?

**Reply by Cra5hedC0w** [reply]
2.3.2

**Reply by weslambert** [reply]
What version Wazuh agent/server are you running?
